### ISSUE TITLE
The second argument in AsDigraph for a transformation

### ISSUE THREAD

**Issue by wilfwilson** [gap-packages/Digraphs#42]
The method for creating a digraph from a transformation is flawed.

Here is the documentation:
```
  If  trans  is  a  transformation  and n is a non-negative integer, then this
  function  returns  the  functional digraph with n vertices defined by trans.
  Specifically, the graph has n edges: for each  vertex  x,  there  is a unique
  edge with source x; this edge has range x^trans.

  If  the  optional  second argument n is not supplied, then the degree of the
  transformation trans is used by default.
```
When `n` is chosen to be the degree of the transformation, or anything higher than that, then all is fine. The problem occurs when `n` is less than the degree. Unless the restriction of `trans` to `[1 .. n]` happens to be a transformation of `[1 .. n]`, then `AsDigraph` returns a corrupt digraph. 

Here is the example from the documentation:
```
gap> f := Transformation([7, 10, 10, 1, 7, 9, 10, 4, 2, 3]);
Transformation( [ 7, 10, 10, 1, 7, 9, 10, 4, 2, 3 ] )
gap> AsDigraph(f); 
<digraph with 10 vertices, 10 edges>
gap> AsDigraph(f, 4); 
<digraph with 4 vertices, 4 edges>
```
The first digraph is fine. However, look at the out-neighbours of the second digraph:
```
gap> OutNeighbours(AsDigraph(f, 4));
[ [ 7 ], [ 10 ], [ 10 ], [ 1 ] ]
```
This is not right, and it's because the transformation does not map the numbers `[1 .. 4]` back to themselves. (And obviously `AsDigraphs` is using `DigraphNC`). If we discard the edges which go to a number higher than 4, then we don't get a functional digraph. So really there's not a sensible way of representing this transformation by a digraph with 4 vertices.

### Suggestions

I suggest that the optional second argument should not be allowed to be smaller than the degree. That would allow the second argument to be greater than or equal to the degree. I then don't see the reason for keeping the second argument at all, since a transformation acts as the identity on any point higher than the degree - not very interesting. But I wouldn't complain about its continued inclusion.

As a side note, this same bug is present in the Semigroups package in `unstable-3.0` in @james-d-mitchell's function `DigraphOfActionOnPoints`, when the optional second argument is too small - and the same idea is relevant to my function `DigraphOfActionOnPairs` which I'm writing.
```
gap> OutNeighbours(DigraphOfActionOnPoints(FullTransformationMonoid(2), 1));
[ [ 1, 2 ] ]
```

**Comment by james-d-mitchell** [comment]
Good catch @wilfwilson, wouldn't it make sense to allow the value of the second parameter `n` to be less than the degree, if the transformation maps `[1 .. n]` to itself?

**Comment by wilfwilson** [comment]
Yes.

**Comment by wilfwilson** [comment]
Fixed by PR #43
